fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2380,6 +2404,12 @@ private void endRecordStoreStateWrite() {
         });
     }
 
+    public static CompletableFuture<RecordMetaDataProto.DataStoreInfo> loadStoreHeaderAsync(@Nonnull FDBRecordContext context, @Nonnull KeySpacePath keySpacePath) {
+        SubspaceProvider subspaceProvider1 = new SubspaceProviderByKeySpacePath(keySpacePath);
+        Subspace subspace1 = subspaceProvider1.getSubspace(context);
+        return readStoreFirstKey(context, subspace1, IsolationLevel.SERIALIZABLE).thenApply(keyValue -> checkAndParseStoreHeader(keyValue, StoreExistenceCheck.NONE, context, subspaceProvider1, subspace1));
```

Comment:
If this takes the `StoreExistenceCheck` as a parameter, acn more of this be shared between the static and non-static version of `loadStoreHeaderAsync`